### PR TITLE
Shoot impl v2

### PR DESCRIPTION
**Pull request by yakisoba0**
Moved MessageBus related classes to new package.
Introduced MessageBuilder and tested it.
Tested Shot event.

Fixed some Checkstyle errors.

### CODE REVIEW

File: src/main/java/com/spanish_inquisition/battleship/client/game/Game.java [hunk 1]
```diff
@@ -59,14 +62,41 @@ public void runTheGame() throws InterruptedException, NullPointerException {
                     continue;
                 }
                 switch (message.getHeader()) {
-                    case PLAYER_TURN: { /*notify about his move */ }
-                    case OPPONENT_TURN: { /* notify that it is not his turn */ }
+                    case PLAYER_TURN: { /*notify about his move */
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/client/game/Game.java]: 👍

File: src/main/java/com/spanish_inquisition/battleship/common/Header.java [hunk 2]
```diff
@@ -18,7 +18,14 @@
     PLAYER_TURN,
     OPPONENT_TURN,
     GAME_WON,
-    MOVE_REGULAR;
+    DECIDE_ON_MOVE,
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/common/Header.java]: 👍

File: src/main/java/com/spanish_inquisition/battleship/server/Player.java [hunk 3]
```diff
@@ -29,4 +30,20 @@ public Fleet getFleet() {
     public boolean hasNoFleet() {
         return fleet.hasNoShips();
     }
+
+    public boolean fleetGotHit(Integer targetedField) {
+        if(fleet.pointIsClaimedByFleet(targetedField)) {
+            return true;
+        }
+
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/server/Player.java]: Ugh that empty line 👎

File: src/main/java/com/spanish_inquisition/battleship/server/bus/MessageBuilder.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
+package com.spanish_inquisition.battleship.server.bus;
+
+import com.spanish_inquisition.battleship.common.Header;
+import com.spanish_inquisition.battleship.common.NetworkMessage;
+import com.spanish_inquisition.battleship.server.fleet.Ship;
+
+public class MessageBuilder {
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/server/bus/MessageBuilder.java]: Builders good 👍

File: src/main/java/com/spanish_inquisition/battleship/server/fleet/FleetParser.java [hunk 5]
```diff
@@ -9,7 +11,7 @@
 
 public class FleetParser {
     public static List<Integer> parseMessageToIntegersList(String message) {
-        String[] messageSplitted = message.split(String.valueOf(':'));
+        String[] messageSplitted = message.split(String.valueOf(NetworkMessage.RESPONSE_HEADER_SPLIT_CHARACTER));
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/server/fleet/FleetParser.java]: 👍

File: src/main/java/com/spanish_inquisition/battleship/server/game_states/ShotState.java [hunk 6]
```diff
@@ -1,46 +1,141 @@
...
+                currentPlayer.getPlayerId(),
+                Header.DECIDE_ON_MOVE.name()
+        );
+
+        if (!shootIfPlayerSentValidMessage(currentPlayer)) {
             return this;
         }
-        return !didPlayerWon(currentPlayer) ?
-                new PlayerActionState(players, requestBus) : new ResultState(players, requestBus);
+        return !didPlayerWon(currentPlayer)
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/server/game_states/ShotState.java]: didPlayerW**i**n

File: src/main/java/com/spanish_inquisition/battleship/server/game_states/ShotState.java [hunk 7]
```diff
@@ -1,46 +1,141 @@
...
+                ? new PlayerActionState(
+                        players, requestBus)
```
Comment (arturAR) [inline src/main/java/com/spanish_inquisition/battleship/server/game_states/ShotState.java]: 👎

**Pull request merged**